src/select.hpp

```diff
@@ -61,7 +61,8 @@ struct i_poll_events;
 class select_base_t : public virtual poller_base_t
 {
   public:
-    typedef fd_t handle_t;
+    typedef void* handle_t;
```

Comment:
Why was this done here? I don't think it is a good idea to use void* internally within libzmq, together with numerous casts that are prone to errors. I know that there are some other uses, but this should not be extended IMO.

In addition, due to the definition of i_poll_events::handle_t, this cannot be anything else than void*. So defining it in every poller implementation is redundant. If the void*-based definition is really added to i_poll_events, it should probably be removed in the poller_t implementations. But also see my comment in i_poll_events itself.

The external C libzmq API has them to make the data structures opaque, but the implementation of libzmq C++, where this should not be necessary.